server/market/orderrouter.go

```diff
@@ -261,16 +272,80 @@ func (r *OrderRouter) handleLimit(user account.AccountID, msg *msgjson.Message)
 	// from the serialized order, is not yet valid. The Market will stamp the
 	// order on receipt, and the order ID will be valid.
 
-	// Send the order to the epoch queue where it will be time stamped.
 	oRecord := &orderRecord{
 		order: lo,
 		req:   limit,
 		msgID: msg.ID,
 	}
-	if err := tunnel.SubmitOrder(oRecord); err != nil {
-		log.Warnf("Market failed to SubmitOrder: %v", err)
-		return msgjson.NewError(msgjson.UnknownMarketError, "failed to submit order")
+
+	swapVal := limit.Quantity
+	lots := swapVal / coins.base.LotSize
+	if !sell {
+		swapVal = matcher.BaseToQuote(limit.Rate, limit.Quantity)
 	}
+
+	checkCoins := func() (tryAgain bool, msgErr *msgjson.Error) {
+		var valSum uint64
+		var spendSize uint32
+		for _, coin := range limit.Trade.Coins {
```

Comment:
Great point.  I'll have the closure capture a list of coins to check that it pares down, although because of the possibility of a reorg happening that makes one previously confirmed coin unconfirmed that could be a little dicey.

Reply:
Nevermind that reorg comment, we're fine with mempool here anyway (not a confirm check).